src/Hedgehog/Extras/Test/Golden.hs

```diff
@@ -69,35 +119,21 @@ diffVsGoldenFile
   => String   -- ^ Actual content
   -> FilePath -- ^ Reference file
   -> m ()
-diffVsGoldenFile actualContent referenceFile = GHC.withFrozenCallStack $ do
+diffVsGoldenFile actualContent goldenFile = GHC.withFrozenCallStack $ do
   forM_ mGoldenFileLogFile $ \logFile ->
-    liftIO $ semBracket $ IO.appendFile logFile $ referenceFile <> "\n"
+    liftIO $ semBracket $ IO.appendFile logFile $ goldenFile <> "\n"
 
-  fileExists <- liftIO $ IO.doesFileExist referenceFile
+  fileExists <- liftIO $ IO.doesFileExist goldenFile
 
-  if fileExists
-    then do
-      referenceLines <- List.lines <$> H.readFile referenceFile
-      let difference = getGroupedDiff actualLines referenceLines
-      case difference of
-        []       -> pure ()
-        [Both{}] -> pure ()
-        _        -> do
-          H.note_ $ "Golden test failed against golden file: " <> referenceFile
-          failMessage callStack $ ppDiff difference
-    else if createFiles
-      then do
-        -- CREATE_GOLDEN_FILES is set, so we create any golden files that don't
-        -- already exist.
-        H.note_ $ "Creating golden file " <> referenceFile
-        H.createDirectoryIfMissing_ (takeDirectory referenceFile)
-        H.writeFile referenceFile actualContent
-      else do
-        H.note_ $ mconcat
-          [ "Golden file " <> referenceFile
-          , " does not exist.  To create, run with CREATE_GOLDEN_FILES=1"
-          ]
-        H.failure
+  if recreateGoldenFiles
+    then writeGoldenFile goldenFile actualContent
+    else if createGoldenFiles
+      then if fileExists
+        then checkAgainstGoldenFile goldenFile actualLines
+        else writeGoldenFile goldenFile actualContent
+      else if fileExists
+        then checkAgainstGoldenFile goldenFile actualLines
+        else reportGoldenFileMissing goldenFile
```

Comment:
I think this might be simpler (with `MultiWayIf`):
```suggestion
  if | fileExists && not recreateGoldenFiles -> checkAgainstGoldenFile goldenFile actualLines
     | (recreateGoldenFiles || (createGoldenFiles && not fileExists)) -> writeGoldenFile goldenFile actualContent
     | not fileExists -> reportGoldenFileMissing goldenFile
     | otherwise -> checkAgainstGoldenFile goldenFile actualLines
```

Reply:
Personally I found the initial version easier to understand :innocent:

Reply:
Oh actually the final version (in https://github.com/input-output-hk/hedgehog-extras/pull/54/commits/2e49c81e43404e6dd2963d191156268439128047) is the best one (`MultiwayIf` but more simple) :+1: